### PR TITLE
Add kubetest flag for passing command to gather cluster logs on GKE

### PR DESCRIPTION
**Pull request by aleksandra-malinowska**
This command is specific to GKE deployer.

### CODE REVIEW

**Comment by aleksandra-malinowska** [comment]
cc @tosi3k

File: kubetest/gke.go [hunk 1]
```diff
@@ -66,6 +66,7 @@ var (
 	gkeDownTimeout                 = flag.Duration("gke-down-timeout", 1*time.Hour, "(gke only) Timeout for gcloud container clusters delete call. Defaults to 1 hour which matches gcloud's default.")
 	gkeRemoveNetwork               = flag.Bool("gke-remove-network", true, "(gke only) At the end of the test remove non-default network that was used by cluster.")
 	gkeDumpConfigMaps              = flag.String("gke-dump-configmaps", "[]", `(gke-only) A JSON description of ConfigMaps to dump as part of gathering cluster logs. Note: --dump or --dump-pre-test-logs flags must also be set. Example: '[{"Name":"my-map", "Namespace":"default", "DataKey":"my-data-key"}]`)
+	gkeDumpLogsCmd                 = flag.String("gke-dump-logs-cmd", "", "(gke-only) if set, run this command to dump cluster logs.")
```
Comment (tosi3k) [inline kubetest/gke.go:69]: Could we rename this flag to sth like `gke-dump-additional-logs` (and hence change the flag description + the log message in `DumpClusterLogs` accordingly)?

In `DumpClusterLogs` function we already collect logs from masters, so such a name would better reflect the flag's semantics.

File: kubetest/gke.go [hunk 2]
```diff
@@ -437,13 +438,25 @@ func (g *gkeDeployer) IsUp() error {
...
 // DumpClusterLogs for GKE generates a small script that wraps
 // log-dump.sh with the appropriate shell-fu to get the cluster
 // dumped.
 //
 // TODO(zmerlynn): This whole path is really gross, but this seemed
 // the least gross hack to get this done.
 func (g *gkeDeployer) DumpClusterLogs(localPath, gcsPath string) error {
+	if *gkeDumpLogsCmd != "" {
+		if err := g.dumpLogs(); err != nil {
+			// Log a warning and carry on
```
Comment (tosi3k) [inline kubetest/gke.go:455]: I think we'd like to throw an error as well - if the additional command fails, then we'd like to see that in the Prow job execution as well.

**Review by tosi3k — CHANGES_REQUESTED** [review]
Please squash the commits into a single one before submitting the PR.

File: kubetest/gke.go (hunk 2 above)
Comment (aleksandra-malinowska) [reply kubetest/gke.go]: Done.

File: kubetest/gke.go (hunk 1 above)
Comment (aleksandra-malinowska) [reply kubetest/gke.go]: Done.

**Comment by aleksandra-malinowska** [comment]
Squashed and rebased.

File: kubetest/gke.go [hunk 3]
```diff
@@ -531,14 +550,19 @@ export KUBE_NODE_OS_DISTRIBUTION='%[3]s'
...
+		errs = append(errs, err)
+		return wrapErrors("DumpClusterLogs", errs...)
 	}
+
 	if err := ioutil.WriteFile(filepath.Join(localPath, "gke-configmap.json"), jsonDump, 0644); err != nil {
-		return err
+		errs = append(errs, err)
+		return wrapErrors("DumpClusterLogs", errs...)
 	}
 	return nil
```
Comment (tosi3k) [inline kubetest/gke.go:567]: We need to check if `len(errs) == 0` here and return appropriate error based on that instead of returning `nil` here.

**Review by tosi3k — CHANGES_REQUESTED** [review]
Would it make sense to change the logic here as well: https://github.com/kubernetes/test-infra/blob/e7676ddfe7efc13684916a84d731ca315189a6d2/kubetest/e2e.go#L118-L125 to return an error if the `DumpClusterLogs` method returns an error as well?

File: kubetest/gke.go (hunk 3 above)
Comment (aleksandra-malinowska) [reply kubetest/gke.go:572]: Done.

**Review by feiskyer — APPROVED** [review]
LGTM

**Pull request merged**